book/basics/units_basics.nim

```diff
@@ -0,0 +1,150 @@
+import nimib, nimibook
+
+nbInit()
+nbUseNimibook
+nbDoc.useLatex
+
+nbText: """
+# Using and checking units with [Unchained](https://github.com/SciNim/Unchained)
+
+Units are crucial in physics, without them we wouldn't have any consistent way of comparing two measurements. 
+You wouldn't solve an equation without checking units afterwards as a safety-check. But when it comes to coding 
+physical formulas units are often ignored or in best cases in a comment somewhere. This can be solved by using
+unit libraries like `Unchained`. What puts `Unchained` apart from many other libraries is that it does the majority
+of the work at compile-time so that you know that if the code compiles, then the units are correct. 
+Other libraries would probably check this at runtime, and it will complain first when it sees the piece of code it wants to check
+the units of. As an added bonus, doing most of the work when compiling the code will make the code run faster as there will at most
+be a conversion factor injected.
+
+## Motivating example
+
+One example of a real disaster caused by conversion errors is the Mars Climate Orbiter that NASA sent to the red planet in 1999. 
+It crashed. Why? Because the navigators at JPL used metric units while the manufacturers had used imperial units. So when
+the navigators thought they read 1 N⋅s from the craft's sensors it was in fact 1 lbf⋅s.
+
+As an introduction to `Unchained`, let's find out what 1 lbf⋅s is in N⋅s to understand how much of a difference it made.
+"""
+nbCode:
+  import unchained
+
+  ## Define composite units
+  defUnit(N•s)
+  defUnit(lbf•s)
+
+  ## Assign variables
+  let lbfs = 1.lbf•s
+  let Ns = lbfs.to(N•s)
+
+  echo lbfs, " equals ", Ns
+
+nbText: """
+As you can see, they thought the value they read was 4.4 times smaller than it really was! 
+
+But what does the code do? Let's dissect it block-by-block:
+
+1. `defUnit(N•s)` - There are a ridiculous amount of combinations of base units.
+Therefore you must define composite units manually before you use them. More on this in a later section.
+2. `let lbfs = 1.lbf•s` - We use the dot (`.`) to assign a unit to a number. In this case we assign a variable with the value "1 lbf•s".
+3. `let Ns = lbfs.to(N•s)` - We use the `to` template to convert a variable from one unit to another.
```

Comment:
Technically `to` is currently a proc. I should probably mark it inline, because it's so simple though. I wanted to keep it as a proc as it rather simple (it just returns `(inputValue * conversionFactor).OutputType` as the result). Not that it really matters here.

Reply:
Oops, want to remember that I saw it as a template while I was looking at the source code. Mandella effect? 🤣